### PR TITLE
Rk second multiroom merge

### PR DESCRIPTION
**Pull request by krzysieksr**

### CODE REVIEW

**Comment by gacikjakub** [comment]
Build 49 Fixed
Deployed under: rk-second-multiroom-merge.battleships.kuba005.nazwa.pl

File: backend/src/main/java/pl/krkteam/battleships/common/domain/cell/MastState.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package pl.krkteam.battleships.common.domain.cell;
 
-public enum  MastState{
+public enum MastState {
```
Comment (JaroslawZaczyk) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/cell/MastState.java:3]: good change! Congratulations :D

File: backend/src/main/java/pl/krkteam/battleships/common/dto/PlayerDTO.java [hunk 2]
```diff
@@ -4,17 +4,15 @@
 
     public String name;
 
-    @Override
-    public String toString() {
-        return "Player: {name: " + name + "}";
+    public PlayerDTO() {
     }
```
Comment (JaroslawZaczyk) [inline backend/src/main/java/pl/krkteam/battleships/common/dto/PlayerDTO.java]: remove default empty constructor and use constructor with parameter e.g. new PlayerDTO("");

File: backend/src/main/java/pl/krkteam/battleships/room/holder/Room.java [hunk 3]
```diff
@@ -56,7 +56,7 @@ private void initializeGame() {
         }
     }
 
-    public synchronized boolean isPlayerBelongToRoom(Player player) {
+    synchronized boolean isPlayerBelongToRoom(Player player) {
         return playerSet.contains(player);
     }
-}
+}
```
Comment (JaroslawZaczyk) [inline backend/src/main/java/pl/krkteam/battleships/room/holder/Room.java]: new line

File: backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    public void createRoomAndExpectGetThisRoom() {
+        String roomName = "SomeRoom";
+
+        Player playerA = new Player("SomePlayerA");
+        roomHolder.createRoom(playerA, roomName);
+
+        assertEquals(roomHolder.getRoom(roomName).getRoomName(), roomName);
+    }
+
+    @Test(expectedExceptions = NoSuchElementException.class)
```
Comment (JaroslawZaczyk) [inline backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java:119]: Leave @Test annotation only here and annotate whole class as @Test

File: backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+        Player playerA = new Player("SomePlayerA");
+        roomHolder.createRoom(playerA, roomName);
+
+        Player playerB = new Player("SomePlayerB");
+
+        assertEquals(roomHolder.createRoom(playerB, roomName).getResult()
+                , new CreateRoomWrongDTO().getResult());
+    }
+}
```
Comment (JaroslawZaczyk) [inline backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java]: new line

File: backend/src/test/java/pl/krkteam/battleships/room/holder/RoomsControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        mockMvc.perform(post("/room/create")
+                .content(roomJson)
+                .param("playerName", playerName))
+                .andExpect(status()
+                        .is2xxSuccessful());
+
+        verify(roomHolder, times(1)).createRoom(any(), anyString());
+    }
+
+}
```
Comment (JaroslawZaczyk) [inline backend/src/test/java/pl/krkteam/battleships/room/holder/RoomsControllerTest.java]: new line

File: backend/src/test/java/pl/krkteam/battleships/wait/opponent/WaitForOpponentControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+        mockMvc.perform(get("/room/opponent/present")
+                .param("playerName", waitingPlayer)
+                .param("roomName", roomName))
+                .andExpect(status()
+                        .is2xxSuccessful());
+
+        verify(roomHolder, times(1)).isOpponentInRoom(anyString(), any());
+    }
+}
```
Comment (JaroslawZaczyk) [inline backend/src/test/java/pl/krkteam/battleships/wait/opponent/WaitForOpponentControllerTest.java]: new line

File: frontend/src/app/rest/get/first-round-asker.service.ts [hunk 8]
```diff
@@ -1,20 +1,21 @@
...
+  hostUrl: string = GetRequestExecutor.getHostString()
   
-  constructor(httpClient: HttpClient, private playersService: PlayersService) {
-    super(httpClient);
+  constructor(httpClient: HttpClient, private playersService: PlayersService, private roomsService: RoomsService) {
+    super(httpClient)
   }
   
   public getFirstRoundResult() {
     //todo REST API turn to round
```
Comment (JaroslawZaczyk) [inline frontend/src/app/rest/get/first-round-asker.service.ts:17]: TODO without expiring date

File: frontend/src/app/rest/get/opponent-present-asker.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+export class OpponentPresentAsker extends GetRequestExecutor {
+  
+  hostUrl: string = GetRequestExecutor.getHostString()
+  
+  constructor(httpClient: HttpClient, private playersService: PlayersService, private roomsService: RoomsService) {
+    super(httpClient)
+  }
+  
+  public getOpponenIsPresent() {
+    //todo REST API turn to round
```
Comment (JaroslawZaczyk) [inline frontend/src/app/rest/get/opponent-present-asker.ts]: same here

File: frontend/src/app/rest/get/room-join-asker.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+export class RoomJoinAsker extends GetRequestExecutor {
+  
+  hostUrl: string = GetRequestExecutor.getHostString()
+  
+  constructor(httpClient: HttpClient, private playersService: PlayersService, private roomsService: RoomsService) {
+    super(httpClient)
+  }
+  
+  public getCanPlayerJoinRoom() {
+    //todo REST API turn to round
```
Comment (JaroslawZaczyk) [inline frontend/src/app/rest/get/room-join-asker.ts]: and here

File: frontend/src/app/views/waiting-rooms/waiting-opponent-register/waiting-opponent-register.component.ts [hunk 11]
```diff
@@ -22,11 +25,26 @@ export class WaitingOpponentRegisterComponent implements OnDestroy {
...
-      clearInterval(this.askerInterval)
+      this.opponentPresentAsker.getOpponenIsPresent()
+          .then(opponentPresentResponse => {
+        
+            let opponentPresentString = opponentPresentResponse['result']
+            let opponentPresent: boolean = this.responseStringToBoolean(opponentPresentString)
+        
+            if (opponentPresent) {
+              this.router.navigate(['/game/fleet/placing'])
+              // clearInterval(this.askerInterval)
```
Comment (JaroslawZaczyk) [inline frontend/src/app/views/waiting-rooms/waiting-opponent-register/waiting-opponent-register.component.ts:36]: comment

File: backend/src/main/java/pl/krkteam/battleships/common/dto/PlayerDTO.java [hunk 12]
```diff
@@ -4,17 +4,15 @@
 
     public String name;
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/common/dto/PlayerDTO.java]: final private, method to get to that field

File: backend/src/main/java/pl/krkteam/battleships/common/dto/ShipFromFronted.java [hunk 13]
```diff
@@ -4,12 +4,12 @@
...
 
-  public String[] ships;
-
-  @Override
-  public String toString() {
-    return "ShipFromFronted{" +
-       "shipDTOS=" + Arrays.toString(ships) +
-       '}';
-  }
+    public String[] ships;
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/common/dto/ShipFromFronted.java:7]: private

File: backend/src/main/java/pl/krkteam/battleships/common/dto/ShipHolderDTO.java [hunk 14]
```diff
@@ -4,22 +4,22 @@
...
-  public String toString() {
-    return "ShipHolderDTO{" +
-       "shipDTOS=" + Arrays.toString(shipArray) +
-       '}';
-  }
-
-  public ShipDTO[] getShipArray() {
-    return shipArray;
-  }
+    private ShipDTO[] shipArray;
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/common/dto/ShipHolderDTO.java:7]: final

File: backend/src/main/java/pl/krkteam/battleships/room/holder/RoomHolder.java [hunk 15]
```diff
@@ -14,41 +14,41 @@
...
+        final Room room = roomNameToRoomMap.get(roomName);
         if (room == null) {
             throw new NoSuchElementException();
         }
         return room;
     }
 
-    public CreateResultDTO createRoom(Player player, String roomName) {
-        if (roomMap.containsKey(roomName)) {
+    CreateResultDTO createRoom(Player player, String roomName) {
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/room/holder/RoomHolder.java]: createRoomWithPlayer ?

File: backend/src/main/java/pl/krkteam/battleships/room/holder/dto/create/room/CreateResultDTO.java [hunk 16]
```diff
@@ -2,4 +2,8 @@
 
 public abstract class CreateResultDTO {
     String result;
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/room/holder/dto/create/room/CreateResultDTO.java:4]: private

File: backend/src/main/java/pl/krkteam/battleships/room/holder/dto/join/result/JoinResultDTO.java [hunk 17]
```diff
@@ -3,4 +3,8 @@
 public abstract class JoinResultDTO {
 
     String result;
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/room/holder/dto/join/result/JoinResultDTO.java:5]: private

File: backend/src/main/java/pl/krkteam/battleships/room/holder/dto/room/list/RoomListDTO.java [hunk 18]
```diff
@@ -3,16 +3,16 @@
 import java.util.Arrays;
 
 public class RoomListDTO {
-    private RoomDTO[] roomName;
+    private RoomDTO[] roomDTOS;
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/room/holder/dto/room/list/RoomListDTO.java:6]: final

File: backend/src/main/java/pl/krkteam/battleships/room/holder/RoomHolder.java (hunk 15 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/room/holder/RoomHolder.java]: Player already exist

File: backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/ShipController.java [hunk 19]
```diff
@@ -83,7 +83,8 @@ private Game getGameFromRoom(String roomName) {
 
     private void resetGameBoardIfValidationFailed(PlacingValidationResultDTO placingValidationResultDTO,
                                                   GameBoard playerGameBoard) {
-        if (placingValidationResultDTO.getResult().equals(PlacingValidationResultDTO.Result.WRONG)) {
+        if (placingValidationResultDTO.getResult()
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/ShipController.java]: train, extract to separate method

File: backend/src/main/java/pl/krkteam/battleships/room/holder/dto/create/room/CreateResultDTO.java (hunk 16 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/room/holder/dto/create/room/CreateResultDTO.java:4]: it's abstract class, so I that field would be private I don't have access for extending classes to it

File: backend/src/main/java/pl/krkteam/battleships/room/holder/dto/join/result/JoinResultDTO.java (hunk 17 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/room/holder/dto/join/result/JoinResultDTO.java:5]: the same

File: backend/src/main/java/pl/krkteam/battleships/room/holder/dto/room/list/RoomListDTO.java (hunk 18 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/room/holder/dto/room/list/RoomListDTO.java:6]: Not for dto

File: backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/services/ShipDirectionChecker.java [hunk 20]
```diff
@@ -6,7 +6,7 @@
 class ShipDirectionChecker {
     static Direction whatDirection(ShipFromJson shipFromJson) {
         CoordinatesFromJson[] coordinatesFromJsonArray = shipFromJson.getCoordinates();
-        if (coordinatesFromJsonArray.length==1){
+        if (coordinatesFromJsonArray.length == 1) {
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/services/ShipDirectionChecker.java]: extract expression "coordinatesFromJsonArray.length == 1" to descriptive method

File: backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/services/ShipDirectionChecker.java [hunk 21]
```diff
@@ -6,7 +6,7 @@
...
             return Direction.ONE_MAST;
         }
         if (coordinatesFromJsonArray[0].getX() + 1 == coordinatesFromJsonArray[1].getX() &&
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/services/ShipDirectionChecker.java]: extract if expression to descriptive method

File: backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/ShipController.java (hunk 19 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/ShipController.java]: doesn't make any sense to extract it to separate method

File: backend/src/main/java/pl/krkteam/battleships/wait/opponent/dto/WaiterResponseDTO.java [hunk 22]
```diff
@@ -2,4 +2,8 @@
 
 public abstract class WaiterResponseDTO {
     String response;
```
Comment (dmuca) [inline backend/src/main/java/pl/krkteam/battleships/wait/opponent/dto/WaiterResponseDTO.java:4]: private

File: backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/services/ShipDirectionChecker.java (hunk 20 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/services/ShipDirectionChecker.java]: doesn't make any sense to extract it

File: backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/services/ShipDirectionChecker.java (hunk 21 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/ships/placing/validation/services/ShipDirectionChecker.java]: dosen't make any sense

File: backend/src/main/java/pl/krkteam/battleships/wait/opponent/dto/WaiterResponseDTO.java (hunk 22 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/wait/opponent/dto/WaiterResponseDTO.java:4]: the same as above - it's in abstract class

File: backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+        roomHolder = new RoomHolder();
+    }
+
+    @Test
+    public void testGetRoomListAndExpectHasTheSameRoomsNames() {
+        Player playerA = new Player("SomePlayerA");
+        Player playerB = new Player("SomePlayerB");
+        List<String> roomNamesList = Arrays.asList("SomeRoom1", "SomeRoom2");
+        roomHolder.createRoom(playerA, roomNamesList.get(0));
+        roomHolder.createRoom(playerB, roomNamesList.get(1));
```
Comment (dmuca) [inline backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java]: Code repeat in many methods below

+        Player playerA = new Player("SomePlayerA");
+        Player playerB = new Player("SomePlayerB");
+        List<String> roomNamesList = Arrays.asList("SomeRoom1", "SomeRoom2");
+        roomHolder.createRoom(playerA, roomNamesList.get(0));
+        roomHolder.createRoom(playerB, roomNamesList.get(1));

File: backend/src/test/java/pl/krkteam/battleships/wait/opponent/WaitForOpponentControllerTest.java [hunk 24]
```diff
@@ -0,0 +1,69 @@
...
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        waitForOpponentController = new WaitForOpponentController(roomHolder);
+
+        mockMvc = MockMvcBuilders.standaloneSetup(waitForOpponentController).build();
+    }
+
+    @Test
+    public void testIsOpponentPresentAndExpectPresent() throws Exception {
```
Comment (dmuca) [inline backend/src/test/java/pl/krkteam/battleships/wait/opponent/WaitForOpponentControllerTest.java:39]: testIsOpponentPresentAndExpectPresent and testIsOpponentPresentAndExpectAbsent

methods are almost the same, DRY it
just  (43 line)    .thenReturn(new OpponentPresentDTO()); differs

File: backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java (hunk 23 above)
Comment (krzysieksr) [reply backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java]: It doesn't repeat in exactly the same same sequence, so it doesn't make any sense to  extract it to data provider

File: backend/src/test/java/pl/krkteam/battleships/wait/opponent/WaitForOpponentControllerTest.java (hunk 24 above)
Comment (krzysieksr) [reply backend/src/test/java/pl/krkteam/battleships/wait/opponent/WaitForOpponentControllerTest.java:39]: They are ALMOST the same but NOT the same. Both tests takes diffrent assumptions

**Review by dmuca — APPROVED** [review]
More clean code guys, TypeScript was not checked during this review:)
Excluding clean code violation project looks good.

File: backend/src/main/java/pl/krkteam/battleships/room/holder/RoomHolder.java (hunk 15 above)
Comment (krzysieksr) [reply backend/src/main/java/pl/krkteam/battleships/room/holder/RoomHolder.java]: createRoomAndJoinPlayer

**Review by JaroslawZaczyk — APPROVED** [review]

**Review by RafalJDev — APPROVED** [review]
To be or not to be

**Pull request merged**